From 3916d36280c85f043571dbb8747962488e2a58fa Mon Sep 17 00:00:00 2001
From: t <tristan75015@gmail.com>
Date: Wed, 7 Oct 2026 03:31:02 +0200
Subject: [PATCH] MFront: physical bounds of the arguments of the material
 properties

The symbols giving the physical bounds of an argument of a material
property were only exported if this argument also had standard bounds, for
all the interfaces of material properties. Both kinds of bounds are now
exported independently, as for the parameters.
---
 mfront/src/CodeGeneratorUtilities.cxx | 32 +++++++++++----------------
 1 file changed, 13 insertions(+), 19 deletions(-)

diff --git a/mfront/src/CodeGeneratorUtilities.cxx b/mfront/src/CodeGeneratorUtilities.cxx
index af0fae212..8b18b825b 100644
--- a/mfront/src/CodeGeneratorUtilities.cxx
+++ b/mfront/src/CodeGeneratorUtilities.cxx
@@ -297,33 +297,27 @@ namespace mfront {
       const mfront::MaterialPropertyDescription& mpd) {
     const auto prec = out.precision();
     out.precision(14);
+    // the bounds and the physical bounds are independent
     for (const auto& v : mpd.inputs) {
       if (v.arraySize == 1u) {
-        if (!v.hasBounds()) {
-          continue;
+        if (v.hasBounds()) {
+          writeBoundsSymbol(out, name, v.getExternalName(), "", v.getBounds());
         }
-        writeBoundsSymbol(out, name, v.getExternalName(), "", v.getBounds());
-        if (!v.hasPhysicalBounds()) {
-          continue;
+        if (v.hasPhysicalBounds()) {
+          writeBoundsSymbol(out, name, v.getExternalName(), "Physical",
+                            v.getPhysicalBounds());
         }
-        writeBoundsSymbol(out, name, v.getExternalName(), "Physical",
-                          v.getPhysicalBounds());
       } else {
         for (auto idx = 0; idx != v.arraySize; ++idx) {
-          if (!v.hasBounds(idx)) {
-            continue;
+          const auto n = v.getExternalName() + "_mfront_index_" +
+                         std::to_string(idx) + "_";
+          if (v.hasBounds(idx)) {
+            writeBoundsSymbol(out, name, n, "", v.getBounds(idx));
           }
-          writeBoundsSymbol(out, name,
-                            v.getExternalName() + "_mfront_index_" +
-                                std::to_string(idx) + "_",
-                            "", v.getBounds(idx));
-          if (!v.hasPhysicalBounds(idx)) {
-            continue;
+          if (v.hasPhysicalBounds(idx)) {
+            writeBoundsSymbol(out, name, n, "Physical",
+                              v.getPhysicalBounds(idx));
           }
-          writeBoundsSymbol(out, name,
-                            v.getExternalName() + "_mfront_index_" +
-                                std::to_string(idx) + "_",
-                            "Physical", v.getPhysicalBounds(idx));
         }
       }
     }
-- 
2.43.0

